src/image/loading_displaying.js

```diff
@@ -77,12 +77,35 @@ p5.prototype.loadImage = function(path, successCallback, failureCallback) {
     pImg.drawingContext.drawImage(img, 0, 0);
     pImg.modified = true;
 
-    if (typeof successCallback === 'function') {
-      successCallback(pImg);
+    //GIF
+    if (path.match(/\.[gif]+$/i)) {
```

Comment:
Technically a server can still serve an gif even if the URL doesn't end with `.gif`. I wonder if there's a better way to know if an image is an animated gif or not at this stage?

Reply:
Yeah I agree this isn't ideal. I spent a few moments fiddling around with other ideas but didn't know how to do this check without having to load every image even non-gifs as an array buffer. Can definitely give it another shot.